docs/automated-testing/performance-testing/load-testing.md

```diff
@@ -26,21 +26,49 @@ There are a number of basic component which are required to carry out a load tes
-5. **Observability** - Determine which metrics need to gathered to gain insight into throughput, latency, resource utilization, etc.
-6. **Scalability** - Determine the amount of scale needed by load generator, workload application, CPU, Memory, and network components needed to achieve testing goals. The use of kubernetes on the cloud can be used to make testing infinitely scalable.
+4. **Agree on test matrix** - Which load test cases should be run for which combinations of input parameters.
+5. **Select the right tool** - Many frameworks exist for load testing so consider if features and limitations are suitable for your needs. (Some popular tools are listed below).  This may also include development of a custom load test client, see Preparation phase below.
+6. **Observability** - Determine which metrics need to gathered to gain insight into throughput, latency, resource utilization, etc.
+7. **Scalability** - Determine the amount of scale needed by load generator, workload application, CPU, Memory, and network components needed to achieve testing goals. The use of kubernetes on the cloud can be used to make testing infinitely scalable.
+
+### Preparation
+The key activity is to replace the end user client with a test bench that simulates one or more instances of the original client.  For standard 3rd party tools it may suffice to configure the existing test UI before initiating the load tests.  
+
+If a custom client is used, code development will be required:
+
+1. **Custom development** - Design for minimal impact/overhead.  Be sure to capture  only those features of the production client that are relevant from a load perspective.   Does it matter if the same test is duplicated, or must the workload be unique for each test?  Can all tests be run under the same user context?
+2. **Test environment** - Create test environment that resembles production environment.  This includes the platform as well as external systems, e.g., data sources.
+3. **Security contexts** - Be sure to have all requisite security contexts for the test environment.  Automation like pipelines may require special setup, e.g., OAuth2 client credential flow instead of auth code flow, because interactive login is replaced by non-interactive.  Allow planning leeway in case admin approval is required for new security contexts.
+4. **Test data strategy** - Make sure that output data format (ascii/binary/...) is compatible with whatever analysis tool is used in the analysis phase.  This also includes storage areas (local/cloud/...), which may trigger new security contexts.  Bear in mind that it may be necessary to collect data from sources external to the application to correlate potential performance issues with the application behavior.  This includes platform and network metrics.  Make sure to collect data that covers analysis needs (statistical measures, distributions, graphs, etc.).
```

Comment:
```suggestion
4. **Test data strategy** - Make sure that output data format (ascii/binary/...) is compatible with whatever analysis tool is used in the analysis phase. This also includes storage areas (local/cloud/...), which may trigger new security contexts. Bear in mind that it may be necessary to collect data from sources external to the application to correlate potential performance issues with the application behavior. This includes platform and network metrics. Make sure to collect data that covers analysis needs (statistical measures, distributions, graphs, etc.).
```